### DISCUSSION TITLE
HTTPUpdate: preserving some parameters during FOTA

### DISCUSSION THREAD

**Discussion by tipo1000** [espressif/arduino-esp32#8008] [Q&A] [answered]
Hello,

Please point me to right direction if this is not the right forum for this question...

When using HTTPUpdate -library to do FOTA is it possible to preserve some parameters (wifi pw, SSID...) during FOTA?

Thanks!

**Comment by me-no-dev** [comment, marked as answer]
They are preserved. You can call just "WiFi.begin()" to connect with the saved credentials

**Reply by tipo1000** [reply]
OK, thanks!

How about if I need to store some application specific information? For example calibration values of sensors. Where/how would I store them and recall after FOTA?

**Reply by me-no-dev** [reply]
You should use the "Preferences" library for that. Stores in NVS and is not overwritten by OTA

**Reply by tipo1000** [reply]
True, it works with with "Preferences" library.
